src/hooks/useInput.js

```diff
@@ -0,0 +1,41 @@
+import { useCallback, useRef, useState } from 'react'
+import { useFocusEnter, useFocusLeave } from '@aragon/ui'
+
+export function useInput(validationFn) {
+  const [value, setValue] = useState('')
+  const [status, setStatus] = useState('empty')
+
+  const inputRef = useRef()
```

Comment:
Should we allow users to pass the ref? I generally use this pattern: https://github.com/aragon/aragon-ui/blob/master/src/components/SidePanel/SidePanel.js#L286-L288